### PR TITLE
drivers: sensor: adxl367: add trigger support

### PR DESCRIPTION
**Pull request by amiclaus**
Add trigger support for the adxl367 driver.

### CODE REVIEW

File: dts/bindings/sensor/adi,adxl367-i2c.yaml [hunk 1]
```diff
@@ -6,3 +6,11 @@ description: ADXL367 3-axis nanopower accelerometer, accessed through I2C bus
...
 
 include: ["i2c-device.yaml", "adi,adxl367-common.yaml"]
+
+properties:
+  int1-gpios:
+    type: phandle-array
+    description: |
+      The INT1 signal defaults to active high as produced by the
+      sensor.  The property value should ensure the flags properly
+      describe the signal that is presented to the driver.
```
Comment (MaureenHelm) [inline dts/bindings/sensor/adi,adxl367-i2c.yaml:16]: Move to `adi,adxl367-common.yaml`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by amiclaus — COMMENTED** [review]
changes in v2:
- move `int1-gpios` property to `adi,adxl367-common.yaml`

**Review by MaureenHelm — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
@teburd @yperess please take a look

File: drivers/sensor/adxl367/adxl367.h [hunk 2]
```diff
@@ -312,4 +352,13 @@ struct adxl367_dev_config {
 int adxl367_spi_init(const struct device *dev);
 int adxl367_i2c_init(const struct device *dev);
 
+#ifdef CONFIG_ADXL367_TRIGGER
+int adxl367_trigger_set(const struct device *dev,
+			const struct sensor_trigger *trig,
+			sensor_trigger_handler_t handler);
+
+int adxl367_init_interrupt(const struct device *dev);
+#endif /* CONFIG_ADT7420_TRIGGER */
```
Comment (yperess) [inline drivers/sensor/adxl367/adxl367.h:361]: There's no need to guard this in the header, if you leave it in, you can make the call site much simpler.

File: drivers/sensor/adxl367/adxl367.c [hunk 3]
```diff
@@ -931,6 +938,14 @@ static int adxl367_probe(const struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_ADXL367_TRIGGER
+	ret = adxl367_init_interrupt(dev);
```
Comment (yperess) [inline drivers/sensor/adxl367/adxl367.c:942]: If you leave these in the header you can use
```c
if (IS_ENABLED(CONFIG_ADXL367_TRIGGER)) {
  ret = adxl367_init_interrupt(dev);
  if (ret != 0) {
    LOG_ERR("Failed to initialize interrupt!");
    return -EIO;
  }
}
```

File: drivers/sensor/adxl367/adxl367_trigger.c [hunk 4]
```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2023 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adxl367
```
Comment (yperess) [inline drivers/sensor/adxl367/adxl367_trigger.c:7]: What is this doing? I'm not seeing a use for this define

File: drivers/sensor/adxl367/adxl367.c [hunk 5]
```diff
@@ -931,6 +938,14 @@ static int adxl367_probe(const struct device *dev)
...
+	if (ret) {
```
Comment (yperess) [inline drivers/sensor/adxl367/adxl367.c:943]: Here and all over this PR, this isn't compliant with the Zephyr coding guidelines, please use an explicit `ret != 0`

See https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#:~:text=The%20controlling%20expression%20of%20an%20if%20statement%20and%20the%20controlling%20expression%20of%20an%20iteration%2Dstatement%20shall%20have%20essentially%20Boolean%20type

And https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_14_04.c#L44

**Review by yperess — CHANGES_REQUESTED** [review]
Overall this looks OK, bu

**Review by amiclaus — COMMENTED** [review]
v3:
- do not guard functions in header file.
- use `IS_ENABLED(CONFIG_ADXL367_TRIGGER)` in adxl367.c
- drop `#define DT_DRV_COMPAT adi_adxl367` from adxl367_trigger.c
- use explicit `ret != 0`

**Review by teburd — DISMISSED** [review]

File: drivers/sensor/adxl367/adxl367_trigger.c [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+		drv_data->drdy_handler = handler;
+		drv_data->drdy_trigger = trig;
+		int_mask = ADXL367_DATA_RDY;
+		break;
+	default:
+		LOG_ERR("Unsupported sensor trigger");
+		return -ENOTSUP;
+	}
+
+	if (handler) {
```
Comment (yperess) [inline drivers/sensor/adxl367/adxl367_trigger.c:113]: handler != NULL

**Review by yperess — CHANGES_REQUESTED** [review]
Sorry, one last nit. I'm going to work on a check for this that will allow others to verify that we're following the coding guidelines via the CI.

**Review by amiclaus — COMMENTED** [review]
v4:
- make explicit condition: `handler != NULL`

**Review by yperess — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**